aiidalab_widgets_base/computers.py

```diff
@@ -0,0 +1,718 @@
+
+    def on_setup_ssh(self, b):
+        """ATTENTION: modifying the order of operations in this function can lead to unexpected problems"""
+        with self._setup_ssh_out:
+            clear_output()
+            self._ssh_keygen()
+
+            #temporary passwords
+            password = self.__password
+            proxy_password = self.__proxy_password
+
+            # step 1: if hostname is not provided - do not do anything
+            if self.hostname is None: # check hostname
+                print("Please specify the computer hostname")
+                return
+
+            # step 2: check if password-free access was enabled earlier
+            if self.can_login():
+                print ("Password-free access is already enabled")
+                return
+
+            # step 3: if can't login already, chek whether all required information is provided
+            if self.username is None: # check username
+                print("Please enter your ssh username")
+                return
+            if len(password.strip()) == 0: # check password
+                print("Please enter your ssh password")
+                return
+
+            # step 4: get the right commands to access the proxy server (if provided)
+            success, proxycmd = self._configure_proxy(password, proxy_password)
+            if not success:
+                return       
+
+            # step 5: modify the ssh config file if necessary
+            if not self.is_in_config():
+                self._write_ssh_config(proxycmd=proxycmd)
+            # TODO: add a check if new config is different from the current one. If so
+            # infrom the user about it.
+            
+            # step 6: make host known by ssh on the proxy server
+            if not self.is_host_known():
+                self._make_host_known(self.hostname,['ssh']+[proxycmd] if proxycmd else [])
+
+            # step 7: sending public key to the main host
+            if not self._send_pubkey(self.hostname, self.username, password):
+                print ("Could not send public key to {}".format(self.hostname))
+                return
+
+            # step 8: final check
+            if self.can_login():
+                print("Automatic ssh setup successful :-)")
+                return
+            else:
+                print("Automatic ssh setup failed, sorry :-(")
+                return
+
+    def on_use_proxy_change(self, b):
+        if self._use_proxy.value:
+            self._proxy_ssh_box.layout.visibility = 'visible'
+        else:
+            self._proxy_ssh_box.layout.visibility = 'hidden'
+            self._use_diff_proxy_username.value = False
+
+    def on_use_diff_proxy_username_change(self, b):
+        if self._use_diff_proxy_username.value:
+            self._proxy_user_password_box.layout.visibility = 'visible'
+        else:
+            self._proxy_user_password_box.layout.visibility = 'hidden'
+    
+    def on_computer_hostname_change(self, b):
```

Comment:
This is nice, but is run every single time a character is typed in or deleted.

Reply:
To best of my knowledge [this](https://ipywidgets.readthedocs.io/en/stable/examples/Widget%20Events.html#Continuous-updates) is the only way to control the behavior.
I tried `continuous_update=False`, but this does not work properly

Reply:
Okay. From your link it also seems this only works for Text and Int widgets. I am also not sure exactly what can be done with the `widgets.link` function.

Reply:
It worked for me when initiating `_inp_computer_hostname` with `contiunous_update=False`. One simply has to stop editing the text field, and then it will be updated.

Reply:
Try to remove the text from the hostname field -> you will still be able to press the `Get from config` button.

Reply:
And then immediately you won't be able to.
The only problem is if, when you press the button for that edge-case it takes the value from the text field to be empty or what it originally was before you removed the test?

Reply:
just to say that after discussion @ltalirz and @CasperWA we decided to completely remove this button, as it's functionality probably not needed.